hal/armv7m/imxrt/10xx/config.h

```diff
@@ -17,20 +17,28 @@
 #ifndef _CONFIG_H_
 #define _CONFIG_H_
 
+#ifndef __ASSEMBLY__
+
 #include "imxrt.h"
 #include "peripherals.h"
 #include "../types.h"
 #include "../../mpu.h"
 #include "../../string.h"
 
+extern void _end(void);
+
+#define ADDR_SYSPAGE (void *)(((addr_t)_end + (SIZE_PAGE - 1)) & ~(SIZE_PAGE - 1))
+
```

Comment:
Adress of the `syspage` should be defined in position independent code using `_end` symbol.